test/integration/patients/worklist/worklist.js

```diff
@@ -2147,6 +2145,182 @@ context('worklist page', function() {
+
+    cy
+      .get('.list-page__filters')
+      .find('[data-all-filters-region]')
+      .should('contain', '1');
+
+    cy
+      .get('@filtersSidebar')
+      .find('.sidebar__heading')
+      .should('contain', '1');
+
+    cy
+      .get('@filtersSidebar')
+      .find('[data-filter-button]')
+      .first()
+      .click();
+
+    cy
+      .get('.picklist__item')
+      .contains('Group One')
+      .click()
+      .wait('@routeFlows');
+
+    cy
+      .get('.list-page__filters')
+      .find('[data-all-filters-region]')
+      .should('contain', '2');
+
+    cy
+      .get('@filtersSidebar')
+      .find('.sidebar__heading')
+      .should('contain', '2');
+
+    cy
+      .get('@filtersSidebar')
+      .find('.js-clear-filters')
+      .click()
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('owned-by_11111-v4'));
+
+        expect(storage.filters.states).to.deep.equal(['22222', '33333']);
+      })
+      .wait('@routeFlows')
+      .itsUrl()
+      .its('search')
+      .should('contain', 'filter[state]=22222,33333');
+
+    cy
+      .get('.list-page__filters')
+      .find('[data-all-filters-region]')
+      .should('not.contain', '2');
+
+    cy
+      .get('@filtersSidebar')
+      .find('.sidebar__heading')
+      .should('not.contain', '2');
+
+    cy
+      .visit('/worklist/shared-by')
+      .wait('@routeFlows')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('shared-by_11111-v4'));
+
+        expect(storage.filters.states).to.deep.equal(['22222', '33333']);
+      })
+      .itsUrl()
+      .its('search')
+      .should('contain', 'filter[state]=22222,33333');
+
+    cy
+      .get('.list-page__filters')
+      .find('[data-all-filters-region]')
+      .click();
+
+    cy
+      .get('@filtersSidebar')
+      .find('[data-states-filters-region]')
+      .find('.fa-square-check')
+      .should('have.length', 2);
+
+    cy
+      .get('@filtersSidebar')
+      .find('[data-states-filters-region]')
+      .should('contain', 'To Do')
+      .should('contain', 'In Progress');
+
+    cy
+      .visit('worklist/done-last-thirty-days')
+      .wait('@routeFlows')
+      .then(() => {
+        const storage = JSON.parse(localStorage.getItem('done-last-thirty-days_11111-v4'));
+
+        expect(storage.filters.states).to.deep.equal(['55555', '66666', '77777']);
+      })
+      .itsUrl()
+      .its('search')
+      .should('contain', 'filter[state]=55555,66666,77777');
+
+    cy
+      .get('.list-page__filters')
+      .find('[data-all-filters-region]')
+      .click();
+
+    cy
+      .get('@filtersSidebar')
+      .find('[data-states-filters-region]')
+      .find('.fa-square-check')
+      .should('have.length', 3);
+
+    cy
+      .get('@filtersSidebar')
+      .find('[data-states-filters-region]')
+      .should('contain', 'Done')
+      .should('contain', 'Unable to Complete')
+      .should('contain', 'THMG Transfered');
+
```

Comment:
Quick overview for what I added to this test:

1. Verify default states are shown in the UI based on local storage setting.
2. Unselect one state (1 of 2) and verify that the API request param data, local storage data, and UI total filter counts are correct.
3. Unselect second state (2 of 2) and verify that the API request param data, local storage data, and UI total filter counts are correct. API route should contain `filter[state]=nil_uuid`.
4. Clear filters via the `Clear Filters` button in the UI. Verify API request param data and local storage data is set to the correct default states. And the UI filter counts are back to `0`.
5. Switch to the `Shared By` worklist (no initial local storage data set) and verify that the correct default states filters are added to local storage.
6. Switch to `Done < 30 Days` worklist and verify that only done states are shown in the UI. And that the API request params and local storage data is correct.